Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -39,16 +39,16 @@ HTMLInputElement::~HTMLInputElement()
 
 RefPtr<Layout::Node> HTMLInputElement::create_layout_node(NonnullRefPtr<CSS::StyleProperties> style)
 {
-    if (type() == "hidden")
+    if (type_state() == TypeAttributeState::Hidden)
```

Comment:
`type_state()` does a set of string comparisons on the possible types to find the enum value on each invocation. So if this type is `radio`, this function alone will do something like 85 comparisons before we get to creating `RadioButton`. Let's only call it once at the beginning of each function.

(`type()` had the same issue, so this isn't new, but the end result of this change isn't really saving anything because of that)

Alternatively, I think we could simply add a hook to `HTMLInputElement::parse_attribute` to parse and cache the `type` attribute. Then `type_state()` can just return that.

Reply:
> Alternatively, I think we could simply add a hook to HTMLInputElement::parse_attribute to parse and cache the type attribute. Then type_state() can just return that.

This is definitely what we should do. :)